ee/localserver/krypto-ec-middleware.go

```diff
@@ -137,3 +127,38 @@ func (e *kryptoEcMiddleware) Wrap(next http.Handler) http.Handler {
 		}
 	})
 }
+
+func extractChallenge(r *http.Request) (*challenge.OuterChallenge, error) {
+	// first check query parmeters
+	rawBox := r.URL.Query().Get("box")
+	if rawBox != "" {
+		decoded, err := base64.StdEncoding.DecodeString(rawBox)
+		if err != nil {
+			return nil, fmt.Errorf("decoding b64 box from url param: %w", err)
+		}
+
+		return challenge.UnmarshalChallenge(decoded)
+	}
+
+	// now check body
+	if r.Body == nil {
+		return nil, fmt.Errorf("no box found in url params or request body: body nil")
+	}
+
+	var body map[string]string
```

Comment:
Do you think `map[string]string` or `map[string]any` or even something structured? I'm kinda good with whatever, but have run into problems with booleans in json being unmarshalled to `map[string]string`

Reply:
switch to any, just in case, add type check to string before decoding b64